### DISCUSSION TITLE
Eloquent new methods for ILIKE in MySql,  TRIM and LOWERCASE selected fields (applicable for strings and JSON_EXTRACTS)

### DISCUSSION THREAD

**Discussion by geanruca** [laravel/framework#52540] [Ideas]
Hello! I would like to contribute two new methods in Eloquent called `selectTrim()`, `selectLower()`, and `selectTrimLower()` to select new fields. I would also like to make them applicable to Jsons. If this is useful, I can do it. 

Also, I would like to add the operator "`ILIKE`", which is native from PostgreSQL, to MySQL.

If any of these features already exists, please let me know :'D

This would be my first contribution to Laravel, so I would like some guidance related to Eloquent.

Thanks for your attention and help!

**Comment by ghost** [comment]
Hi. A package would be faster and better to make, via macros.

**Reply by geanruca** [reply]
I don´t think so because the macros are applied to the collection, which is created after getting the values from the query.

**Reply by ghost** [reply]
namespace Illuminate\Database\Query;

class Builder
{
    use BuildsQueries, ExplainsQueries, ForwardsCalls, Macroable {
        __call as macroCall;
    }
...
 
The eloquent builder forwards calls to query builder which is macroable.

**Reply by ghost** [reply]
I use macros for both eloquent and query builder.
EDIT. So, please think again.

**Reply by macropay-solutions** [reply]
Example https://github.com/macropay-solutions/laravel-crud-wizard-free/blob/9cb27b576cf8d1bb462296fcf6cf7b14d18c368c/src/Providers/CrudProvider.php#L23
